src/main/java/teammates/common/datatransfer/attributes/FeedbackResponseAttributes.java

```diff
@@ -44,53 +44,128 @@
-        this.recipientSection = fr.getRecipientSection() == null ? Const.DEFAULT_SECTION : fr.getRecipientSection();
-        this.responseMetaData = fr.getResponseMetaData();
-        this.createdAt = fr.getCreatedAt();
-        this.updatedAt = fr.getUpdatedAt();
-    }
-
-    public FeedbackResponseAttributes(FeedbackResponseAttributes copy) {
-        this.feedbackResponseId = copy.getId();
-        this.feedbackSessionName = copy.feedbackSessionName;
-        this.courseId = copy.courseId;
-        this.feedbackQuestionId = copy.feedbackQuestionId;
-        this.feedbackQuestionType = copy.feedbackQuestionType;
-        this.giver = copy.giver;
-        this.giverSection = copy.giverSection;
-        this.recipient = copy.recipient;
-        this.recipientSection = copy.recipientSection;
-        this.responseMetaData = copy.responseMetaData;
-        this.createdAt = copy.createdAt;
-        this.updatedAt = copy.updatedAt;
+    protected FeedbackResponseAttributes() {
+        // Empty constructor for Builder to construct the object
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static FeedbackResponseAttributes valueOf(FeedbackResponse fr) {
+        return builder()
+                .withFeedbackResponseId(fr.getId())
+                .withFeedbackSessionName(fr.getFeedbackSessionName())
+                .withCourseId(fr.getCourseId())
+                .withFeedbackQuestionId(fr.getFeedbackQuestionId())
+                .withFeedbackQuestionType(fr.getFeedbackQuestionType())
+                .withGiver(fr.getGiverEmail())
+                .withGiverSection(fr.getGiverSection() == null ? Const.DEFAULT_SECTION : fr.getGiverSection())
+                .withRecipient(fr.getRecipientEmail())
+                .withRecipientSection(fr.getRecipientSection() == null ? Const.DEFAULT_SECTION : fr.getRecipientSection())
+                .withCreatedAt(fr.getCreatedAt())
+                .withUpdatedAt(fr.getUpdatedAt())
+                .withResponseMetaData(fr.getResponseMetaData())
+                .build();
+    }
+
+    public FeedbackResponseAttributes getCopy() {
+        return builder()
+                .withFeedbackResponseId(feedbackResponseId)
+                .withFeedbackSessionName(feedbackSessionName)
+                .withCourseId(courseId)
+                .withFeedbackQuestionId(feedbackQuestionId)
+                .withFeedbackQuestionType(feedbackQuestionType)
+                .withGiver(giver)
+                .withGiverSection(giverSection)
+                .withRecipient(recipient)
+                .withRecipientSection(recipientSection)
+                .withResponseMetaData(responseMetaData)
+                .withCreatedAt(createdAt)
+                .withUpdatedAt(updatedAt)
+                .build();
+    }
+
+    /**
+     * A Builder class for {@link FeedbackResponseAttributes}.
+     */
+
+    public static class Builder {
```

Comment:
Yeah, I know the same concern will appear in the entity classes as well but since we are already this far along following this approach of builders we can address them later right?